### DISCUSSION TITLE
exploring-dotnet-6-part-11-callerargumentexpression-and-throw-helpers/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#21] [Announcements]
# exploring-dotnet-6-part-11-callerargumentexpression-and-throw-helpers/

In this post I describe the new C# 10 attribute, [CallerArgumentExpression], and show how it's used in .NET 6 to create a throw helper for ArgumentNullException

https://andrewlock.net/exploring-dotnet-6-part-11-callerargumentexpression-and-throw-helpers/

**Comment by harutyunyangg** [comment]


**Reply by andrewlock** [reply]
Which part don't you understand? 🙂 Having a `throw` in your method makes inclining less likely. Moving it to a helper means your method does not have a `throw`

**Reply by harutyunyangg** [reply]
Dear Andrew, thank you for clarifying, logic is understandable, but the sentence was unclear.

**Comment by moonymachine** [comment]
Thank you for writing this. I can't wait wait to try out the ArgumentNullException throw helper. I am going to simplify my DLL interdependencies by reducing the need for an exception thrower across libraries.
